### ISSUE TITLE
Second derivative, calling autograd.grad twice causes and error? 

### ISSUE THREAD

**Issue by Mehdishishehbor** [NVlabs/tiny-cuda-nn#156]
Hi,
Thanks for your inspiring research. I am trying to get a second derivative of output with respect to input after passing the input to the **tcnn.NetworkWithInputEncoding** wish **hashtable** encoding.

I use the code below to get he second derivitive and loss is defined from the output of tcnn.NetworkWithInputEncoding.
```
first_derivative = autograd.grad(loss, x, create_graph=True)[0]
# We now have dloss/dx
second_derivative = autograd.grad(first_derivative, x)[0]
# This computes d/dx(dloss/dx) = d2loss/dx2
  
```
However I get the following error: 
**_RuntimeError: DifferentiableObject::backward_backward_input_impl: not implemented error_**

Could please help me to understand this error and how I can fix it please? 

I appreciate your time and efforts!
Thanks,
Mehdi

**Comment by Tom94** [comment]
Hi there, second derivatives are not supported by the __tiny-cuda-nn__'s neural networks. Only by the hash encoding. If you'd like to use second derivatives, you need to compose the hash encoding with PyTorch-side neural network (i.e. use `tcnn.Encoding` and `torch.nn.Linear`  rather than `tcnn.NetworkWithInputEncoding`).

More info here: https://github.com/NVlabs/tiny-cuda-nn/issues/131#issuecomment-1210726647

Cheers

**Comment by Mehdishishehbor** [comment]
Thank you!
